TypeDB.java

```diff
@@ -21,41 +21,26 @@
 
 package com.vaticle.typedb.client;
 
-import com.vaticle.typedb.client.api.TypeDBClient;
+import com.vaticle.typedb.client.api.TypeDBConnection;
 import com.vaticle.typedb.client.api.TypeDBCredential;
-import com.vaticle.typedb.client.connection.cluster.ClusterClient;
-import com.vaticle.typedb.client.connection.core.CoreClient;
+import com.vaticle.typedb.client.connection.TypeDBConnectionImpl;
 
 import java.util.Set;
 
 import static com.vaticle.typedb.common.collection.Collections.set;
 
 public class TypeDB {
-
     public static final String DEFAULT_ADDRESS = "localhost:1729";
 
-    public static TypeDBClient coreClient(String address) {
-        return new CoreClient(address);
-    }
-
-    public static TypeDBClient coreClient(String address, int parallelisation) {
-        return new CoreClient(address, parallelisation);
-    }
-
-    public static TypeDBClient.Cluster clusterClient(String address, TypeDBCredential credential) {
-        return new ClusterClient(set(address), credential);
-    }
-
-    public static TypeDBClient.Cluster clusterClient(String address, TypeDBCredential credential, int parallelisation) {
-        return new ClusterClient(set(address), credential, parallelisation);
+    public static TypeDBConnection coreClient(String address) {
```

Comment:
I think we should continue to use `TypeDBClient`, since in Java the 'model' is still that it goes Client -> Session -> Transaction. In Rust we have `DatabaseManager(connection)`, so it's a different model.